tests/Http/PostTest.php

```diff
@@ -1139,6 +1139,38 @@ public function testPostsIndexAsOwnerHiddenPosts()
         $response->assertJsonCount(3, 'data');
     }
 
+    /**
+     * Test for retrieving voter registration posts as a referrer.
+     * A referrer should see all completed voter registrations that they have referred.
+     *
+     * GET /api/v3/posts
+     * @return void
+     */
+    public function testPostsIndexAsVoterRegistrationReferrer()
+    {
+        $referrerUserId = $this->faker->unique()->northstar_id;
+
+        // Add two completed voter reg referrals for our referrer, which should be visible to them.
+        factory(Post::class)->states('voter-reg', 'register-form')->create(['referrer_user_id' => $referrerUserId]);
+        factory(Post::class)->states('voter-reg', 'register-OVR')->create(['referrer_user_id' => $referrerUserId]);
+
+        // Add a non-completed voter reg referral for our referrer, which shouldn't be visible.
+        factory(Post::class)->states('voter-reg', 'step-1')->create(['referrer_user_id' => $referrerUserId]);
+
+        // Add a completed voter reg without a referrer, which shouldn't be visible.
+        factory(Post::class)->states('voter-reg', 'register-OVR')->create();
+
+        // Add a completed voter reg referral for a different referrer, which shouldn't be visible.
+        factory(Post::class)->states('voter-reg', 'register-form')->create(['referrer_user_id' => $this->faker->unique()->northstar_id]);
+
+        // Add a pending photo post, which shouldn't be visible.
+        factory(Post::class)->states('photo', 'pending')->create();
+
+        $response = $this->withAccessToken($referrerUserId)->getJson('api/v3/posts');
+        $response->assertStatus(200);
+        $response->assertJsonCount(2, 'data');
```

Comment:
Do you think we need to make sure the 2 posts visible here are the correct 2? I feel like odds that something breaks so the wrong 2 posts are showing up are pretty slim but in terms of covering all our bases...

Reply:
I was following by example from the other tests, but agree that it couldn't hurt! Updated this branch with [`assertJsonFragment`](https://laravel.com/docs/5.7/http-tests#assert-json-fragment) checks for the expected post ID's.

Reply:
@aaronschachter looks awesome!!